test/e2e/pvc_label_selector.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+	vsphere "k8s.io/kubernetes/pkg/cloudprovider/providers/vsphere"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+/*
+   This is a function test for Selector-Label Volume Binding Feature
+   Test verifies volume with the matching label is bounded with the PVC.
+
+   Test Steps
+   ----------
+   1. Create VMDK.
+   2. Create pv with lable volume-type:ssd, volume path set to vmdk created in previous step, and PersistentVolumeReclaimPolicy is set to Delete.
+   3. Create PVC (pvc_vvol) with label selector to match with volume-type:vvol
+   4. Create PVC (pvc_ssd) with label selector to match with volume-type:ssd
+   5. Wait and verify pvc_ssd is bound with PV.
+   6. Verify Status of pvc_vvol is still pending.
+   7. Delete pvc_ssd.
+   8. verify associated pv is also deleted.
+   9. delete pvc_vvol
+
+*/
+var _ = framework.KubeDescribe("pvclabelselector", func() {
```

Comment:
Similar to the new pvc_reclaimpolicy.go, I think the outer `KubeDescribe` should contain the string "PersistentVolumes" so that a) this test is run with `focus=PersistentVolumes`, and b) so that the label selector test suite is generic enough for other volumes besides vsphere. An inner `KubeDescribe` should have the "vsphere" string.